src/control.lua

```diff
@@ -361,113 +370,101 @@ end
 --[[Getter and setter update methods]]--
 ----------------------------------------
 function ResetRequestGathering()
-	RestartIterator(global.outputChestsData.entitiesData     , TICKS_TO_COLLECT_REQUESTS)
+	global.outputChestsData.entitiesData.pos = 0
 	global.outputChestsData.requests = {}
 
-	RestartIterator(global.outputTanksData.entitiesData      , TICKS_TO_COLLECT_REQUESTS)
+	global.outputTanksData.entitiesData.pos = 0
 	global.outputTanksData.requests = {}
 
-	RestartIterator(global.outputElectricityData.entitiesData, TICKS_TO_COLLECT_REQUESTS)
+	global.outputElectricityData.entitiesData.pos = 0
 	global.outputElectricityData.requests = {}
 end
 
 function ResetFulfillRequestIterators()
-	RestartIterator(global.outputChestsData.requestsLL     , TICKS_TO_FULFILL_REQUESTS)
-	RestartIterator(global.outputTanksData.requestsLL      , TICKS_TO_FULFILL_REQUESTS)
-	RestartIterator(global.outputElectricityData.requestsLL, TICKS_TO_FULFILL_REQUESTS)
+	global.outputChestsData.requestsLL.pos = 0
+	global.outputTanksData.requestsLL.pos = 0
+	global.outputElectricityData.requestsLL.pos = 0
 end
 
 function ResetPutterIterators()
-	RestartIterator(global.inputChestsData.entitiesData     , TICKS_TO_COLLECT_REQUESTS + TICKS_TO_FULFILL_REQUESTS)
-	RestartIterator(global.inputTanksData.entitiesData      , TICKS_TO_COLLECT_REQUESTS + TICKS_TO_FULFILL_REQUESTS)
-	RestartIterator(global.inputElectricityData.entitiesData, TICKS_TO_COLLECT_REQUESTS + TICKS_TO_FULFILL_REQUESTS)
+	global.inputChestsData.entitiesData.pos = 0
+	global.inputTanksData.entitiesData.pos = 0
+	global.inputElectricityData.entitiesData.pos = 0
 end
 
 function PrepareToFulfillRequests()
-	global.outputChestsData.requestsLL      = ArrayToLinkedListOfRequests(global.outputChestsData.requests     , true)
-	global.outputTanksData.requestsLL       = ArrayToLinkedListOfRequests(global.outputTanksData.requests      , false)
-	global.outputElectricityData.requestsLL = ArrayToLinkedListOfRequests(global.outputElectricityData.requests, false)
+	global.outputChestsData.requestsLL      = PrepareRequests(global.outputChestsData.requests     , true)
+	global.outputTanksData.requestsLL       = PrepareRequests(global.outputTanksData.requests      , false)
+	global.outputElectricityData.requestsLL = PrepareRequests(global.outputElectricityData.requests, false)
 end
 
-function RetrieveGetterRequests(allowedToGetElectricityRequests)
-	local chestLL = global.outputChestsData.entitiesData
-	for i = 1, chestLL.iterator.linksPerTick do
-		local nextLink = NextLink(chestLL)
-		if nextLink ~= nil then
-			GetOutputChestRequest(global.outputChestsData.requests, nextLink.data)
+-- Iterates through a sequence over a number of separate runs
+function partial_ipairs(list, runs_left)
+	function iterator(state, pos)
+		if pos >= state.endpoint or pos >= #state.list then
```

Comment:
Endpoint/pos might be past the end of the list if it shrunk between invocations.  It might be better to make a partial_remove function that correctly updates the pos property.